src/components/composite/common/CategorySelector/CategorySelector.tsx

```diff
@@ -0,0 +1,123 @@
+import {
+  alpha,
+  Box,
+  List,
+  ListItemButton,
+  ListItemIcon,
+  ListItemProps,
+  ListItemText,
+  styled,
+  Typography,
+} from '@mui/material';
+import React, { ComponentType, FC, useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+
+export interface CategoryConfig {
+  title: string;
+  icon?: ComponentType;
+}
+
+interface CategorySelectorProps {
+  categories: CategoryConfig[];
+  value?: string;
```

Comment:
I'd put `string | undefined` here. Or even better, `string | null` to explicitly highlight that it's a special value (no category selected) rather than "value not set".